src/transport/transport.ts

```diff
@@ -1,9 +1,11 @@
+import { SocksProxy } from 'socks';
+
 export abstract class Transport<T> {
     /**
      * This will be set by the MqttClient
      */
     public callbacks: TransportConnectOptions;
-    public constructor(protected options: T) {}
+    public constructor(protected options: T, protected proxyOptions?: TransportProxyOptions) {}
```

Comment:
The `proxyOptions` should be in the `options` object for a given `Transport` as not every transport supports a proxy.